dohproxy/utils.py

```diff
@@ -21,16 +21,48 @@
 from dohproxy import constants, server_protocol, __version__
 
 
-def dnsmsg2log(msg: dns.message.Message) -> str:
-    """ Helper function to return a printable excerpt from a dns message object.
-    It handles basic things like potentially empty quesiton section.
+def dnsquery2log(msg: dns.message.Message) -> str:
+    """ Helper function to return a readable excerpt from a dns query object.
     """
-    question = msg.question[0] if len(msg.question) else '<empty>'
-    return 'ID [{}] RCODE [{}] FLAGS [{}] QUESTION [{}]'.format(
+    question = '<empty>'
+    if len(msg.question):
+        questions = str(msg.question[0]).split(' ')
+        qname = questions[0]
+        qclass = questions[1]
+        qtype = questions[2]
+        question = ' '.join([qname, qtype, qclass])
+    flags = '/'.join(dns.flags.to_text(msg.flags).split(' '))
+
+    return '{} {} {}'.format(
+        question,
         msg.id,
-        dns.rcode.to_text(msg.rcode()),
-        dns.flags.to_text(msg.flags),
+        flags,
+    )
+
+
+def dnsans2log(msg: dns.message.Message) -> str:
+    """ Helper function to return a readable excerpt from a dns answer object.
+    """
+    question = '<empty>'
+    if len(msg.question):
+        questions = str(msg.question[0]).split(' ')
+        qname = questions[0]
+        qclass = questions[1]
+        qtype = questions[2]
+        question = ' '.join([qname, qtype, qclass])
+    flags = '/'.join(dns.flags.to_text(msg.flags).split(' '))
+
+    return '{} {} {} {}/{}/{} {}/{}/{} {}'.format(
         question,
+        msg.id,
+        flags,
+        len(msg.answer[0].items) if len(msg.answer) else 0,
```

Comment:
this is not correct, I believe this should be:
```
sum(len(x) for x in msg.answer)
```

The same applies for all sections, so feel free to use a helper function to sum items in a section.